collector/zfs_solaris.go

```diff
@@ -0,0 +1,329 @@
+	arcstatsHits                 *prometheus.Desc
+	arcstatsMisses               *prometheus.Desc
+	arcstatsMFUGhostHits         *prometheus.Desc
+	arcstatsMFUGhostSize         *prometheus.Desc
+	arcstatsMFUSize              *prometheus.Desc
+	arcstatsMRUGhostHits         *prometheus.Desc
+	arcstatsMRUGhostSize         *prometheus.Desc
+	arcstatsMRUSize              *prometheus.Desc
+	arcstatsOtherSize            *prometheus.Desc
+	arcstatsP                    *prometheus.Desc
+	arcstatsSize                 *prometheus.Desc
+	zfetchstatsHits              *prometheus.Desc
+	zfetchstatsMisses            *prometheus.Desc
+}
+
+const (
+	zfsCollectorSubsystem = "zfs"
+)
+
+func init() {
+	registerCollector("zfs", defaultEnabled, NewZfsCollector)
+}
+
+func NewZfsCollector() (Collector, error) {
+	return &zfsCollector{
+		abdstatsLinearCount: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, zfsCollectorSubsystem, "abdstats_linear_count_total"),
+			"ZFS ARC buffer data linear count", nil, nil,
+		),
+		abdstatsLinearDataSize: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, zfsCollectorSubsystem, "abdstats_linear_data_bytes"),
+			"ZFS ARC buffer data linear data size", nil, nil,
+		),
+		abdstatsScatterChunkWaste: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, zfsCollectorSubsystem, "abdstats_scatter_chunk_waste_bytes"),
+			"ZFS ARC buffer data scatter chunk waste", nil, nil,
+		),
+		abdstatsScatterCount: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, zfsCollectorSubsystem, "abdstats_scatter_count_total"),
+			"ZFS ARC buffer data scatter count", nil, nil,
+		),
+		abdstatsScatterDataSize: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, zfsCollectorSubsystem, "abdstats_scatter_data_bytes"),
+			"ZFS ARC buffer data scatter data size", nil, nil,
+		),
+		abdstatsStructSize: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, zfsCollectorSubsystem, "abdstats_struct_bytes"),
+			"ZFS ARC buffer data struct size", nil, nil,
+		),
+		arcstatsAnonSize: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, zfsCollectorSubsystem, "arcstats_anon_bytes"),
+			"ZFS ARC anon size", nil, nil,
+		),
+		arcstatsC: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, zfsCollectorSubsystem, "arcstats_c_bytes"),
+			"ZFS ARC target size", nil, nil,
+		),
+		arcstatsCMax: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, zfsCollectorSubsystem, "arcstats_c_max_bytes"),
+			"ZFS ARC maximum size", nil, nil,
+		),
+		arcstatsCMin: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, zfsCollectorSubsystem, "arcstats_c_min_bytes"),
+			"ZFS ARC minimum size", nil, nil,
+		),
+		arcstatsDataSize: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, zfsCollectorSubsystem, "arcstats_data_bytes"),
+			"ZFS ARC data size", nil, nil,
+		),
+		arcstatsDemandDataHits: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, zfsCollectorSubsystem, "arcstats_demand_data_hits"),
```

Comment:
Are hits/misses counters? If use, they should contain `_total` in the name.

Reply:
Added.